### DISCUSSION TITLE
Extending support for JavaScript?

### DISCUSSION THREAD

**Discussion by leftmostcat** [microsoft/vscode-discussions#577] [Extension Development QnA] [answered]
I work on a project which, in some cases, uses internally-defined API for JavaScript imports or a form of typecasting. Is it possible to query the built-in JS IntelliSense analysis in order to implement an additional provider? I haven't seen anything in the API documentation, but wasn't sure if I'd missed something.

**Comment by r3m0t** [comment, marked as answer]
You can implement a typescriptserverplugin as tsserver provides the javascript support.

**Reply by leftmostcat** [reply]
Unfortunately, it appears that `tsserver` does not allow for modifying the AST between parsing and type checking, which makes it impossible to adjust for custom import API.
